### PR TITLE
Adding incremental test 10.

### PR DESCRIPTION
**Pull request by janciesko**

### CODE REVIEW

File: core/unit_test/incremental/Test10_HierarchicalBasics.hpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+template <class ExecSpace>
+struct HierarchicalBasics {
+  typedef Kokkos::TeamPolicy<ExecSpace> TeamPolicy;
+  typedef Test::Functor<ExecSpace> Functor;
+
+  void run() {
+    int error = 0;
+    Kokkos::parallel_reduce("Teams", TeamPolicy(N, M), Functor(), error);
+    Kokkos::fence();
```
Comment (dalg24) [inline core/unit_test/incremental/Test10_HierarchicalBasics.hpp]: Fence is unnecessary here.

File: core/unit_test/incremental/Test10_HierarchicalBasics.hpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+typedef int value_type;
+
+template <class ExecSpace>
+struct Functor {
+  typedef Kokkos::TeamPolicy<ExecSpace> TeamPolicy;
+  typedef typename TeamPolicy::member_type member_type;
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const member_type& team, value_type& update) const {
+    int m = (int)(team.league_rank() + team.team_rank());
```
Comment (dalg24) [inline core/unit_test/incremental/Test10_HierarchicalBasics.hpp]:
```suggestion
    int m = team.league_rank() + team.team_rank();
```

**Comment by masterleinad** [comment]
```
/var/jenkins/workspace/Kokkos/core/unit_test/incremental/Test10_HierarchicalBasics.hpp(87): error: namespace "Kokkos" has no member "OpenMP"
```

File: core/unit_test/incremental/Test10_HierarchicalBasics.hpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+template <class ExecSpace>
+struct HierarchicalBasics {
+  typedef Kokkos::TeamPolicy<ExecSpace> TeamPolicy;
+  typedef Test::Functor<ExecSpace> Functor;
+
+  void run() {
+    int error = 0;
+    Kokkos::parallel_reduce("Teams", TeamPolicy(N, M), Functor(), error);
```
Comment (crtrott) [inline core/unit_test/incremental/Test10_HierarchicalBasics.hpp]: According to https://github.com/kokkos/internal-documents/blob/master/developer-notes/Backend-Development-Guide.md this should only happen in Test 15. Test 10 should only to parallel for and NO nested things or reductions.

File: core/unit_test/incremental/Test10_HierarchicalBasics.hpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  void run(const int nP, int nT) {
+    if (nT > ExecSpace::concurrency()) nT = ExecSpace::concurrency();
+
+    Kokkos::View<int **, ExecSpace> v("Array_A", nP, nT);
+    Kokkos::parallel_for(
+        "Teams", TeamPolicy(nP, nT), KOKKOS_LAMBDA(const MemberType &team) {
+          const int tR = team.team_rank();
+          const int tS = team.team_size();
+          const int lR = team.league_rank();
+          const int lS = team.league_size();
```
Comment (crtrott) [inline core/unit_test/incremental/Test10_HierarchicalBasics.hpp]: unused variable.

File: core/unit_test/incremental/Test10_HierarchicalBasics.hpp (hunk 1 above)
Comment (crtrott) [reply core/unit_test/incremental/Test10_HierarchicalBasics.hpp]: Actually it might be necessary

**Pull request merged**